litestar/utils/signature.py

```diff
@@ -131,6 +131,24 @@ def _is_any_optional(origin_: Any, args_: tuple[Any, ...]) -> bool:
     return hints
 
 
+def expand_type_var_in_type_hint(type_hint: dict[str, Any], namespace: dict[str, Any] | None) -> dict[str, Any]:
+    """Expand TypeVar for any parameters in type_hint
+
+    Args:
+        type_hint: mapping of parameter to type obtained from calling `get_type_hints` or `get_fn_type_hints`
+        namespace: mapping of TypeVar to concrete type
+
+    Returns:
+        type_hint with any TypeVar parameter expanded
+    """
+    if namespace:
+        expanded_type_hint = {}
+        for name, hint in type_hint.items():
+            expanded_type_hint[name] = _substitute_typevars(hint, namespace)
+        return expanded_type_hint
+    return type_hint
```

Comment:
```suggestion
    if namespace:
        return {
            name: _substitute_typevars(hint, namespace)
            for name, hint in type_hint.items()
        }
    return type_hint
```
maybe be worse or better, who knows 🙃

Reply:
Thanks for the suggestion.